### PR TITLE
Change PSR-0 to PSR-4

### PR DESCRIPTION
**Pull request by trupedia**
Since PSR-0 is deprecated, I changed it to PSR-4 and optimized the testsuite to use the composer generated autoload file (also added phpunit to the required-dev part).
But maybe it is even better just to use the classmap option in composer.json, because it's just a single class without a namespace at all.
Just my thoughts... 

Cheers

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
You should NOT move the file. That's a BC break. You can use PSR4 without moving any files.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/vendor
+/.idea
+composer.lock
```
Comment (GrahamCampbell) [inline .gitignore]: cs

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+/vendor
+/.idea
```
Comment (GrahamCampbell) [inline .gitignore]: NO. This should not go i here. This belongs in your global .gitignore file. People always do this wrong. :(

File: composer.json [hunk 3]
```diff
@@ -12,7 +12,10 @@
             "homepage": "http://erusev.com"
         }
     ],
+    "require-dev": {
+      "phpunit/phpunit": "~4.0"
```
Comment (GrahamCampbell) [inline composer.json]: Also wrong. This should NOT be added. Belongs in your global system install.

**Comment by trupedia** [comment]
@GrahamCampbell thanks for the hints. Learnt a lot :)
Won't happen again!

File: .gitignore (hunk 2 above)
Comment (hkdobrev) [inline .gitignore]: @GrahamCampbell :+1:

**Comment by avindra** [comment]
+1 What's the possibility of switching to psr-4?

**Comment by erusev** [comment]
@avindra it seems like a good idea, so i'd say that there's a good possibility

**Comment by arcanedev-maroc** [comment]
Is there any progress about this issue ?

I've a suggestion about it and we can start applying the PSR-4 autoloading for the next major release `v2`:

 * The namespace will be `Parsedown\Parsedown` for the main class (Parsedown class will be inside the `src` folder).
 * Renaming the `test` folder to `tests` and of course we're going to use the PSR-4 `autoload-dev` and the namespace for the folder is going to be `Parsedown\Tests`.

Something like this:

```json
{
    "autoload": {
        "psr-4": {
            "Parsedown\\": "src/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "Parsedown\\Tests\\": "tests/"
        }
    },
}
```

@erusev , can you create a `v2` branch to test it out ?

**Comment by erusev** [comment]
@arcanedev-maroc i've been busy working on the Windows and Linux version of [Caret](http://caret.io). Hopefully, I'll have a look at this as soon as we release these versions later this month.

**Pull request closed without merging**